### PR TITLE
display: add a non_interactive config parameter

### PR DESCRIPTION
**Pull request by fsouza**
Following up on #106: the idea is to be able to run PackerSync and have
it cleanup stale plugins without asking for input (imagine the user is
in --headless mode).

Originally, I had a no_input parameter, but based on feedback I updated
the code to have a config parameter that modifies the behavior of the
display module.

### CODE REVIEW

**Comment by wbthomason** [comment]
Thanks! This looks good.

I think we can cover most use cases with this option, so I'm happy to merge unless there's more you think should be added. I may eventually want to expand `no_input` to a `no_display` type of option, which would imply `no_input` but also prevent the display window for update and install operations from displaying. This would be to serve both users who run headless operations as well as users who prefer not to have a display window (there's been at least one issue asking about this).

I think that's probably best in another PR, though. If you agree, I'll go ahead and merge. Thanks again!

**Comment by fsouza** [comment]
@wbthomason ohh that's a good call, I can turn this into a `no_display` option later today if you prefer. I can do that before we merge, just to make sure we don't introduce an option then remove it. What do you think?

**Comment by wbthomason** [comment]
If you're willing, that'd be great before we merge. Note that `no_display` is probably easiest implemented as part of the `display` config, since many functions expect a display object. So, I think it should be simplest to make `no_display` control the `display` module to return a display object that skips all operations.

This could probably be done easily by adding a check to avoid opening the window/buffer in the `display.open` function, then adding a check in the `valid_display` method.

If you don't get to this today, no worries. Thanks again!

**Comment by fsouza** [comment]
Sorry @wbthomason didn't get to it yesterday, but will do today.

**Comment by fsouza** [comment]
This should be good for review now @wbthomason!

I changed my config to use packer.nvim from my fork and was able to test PackerSync, PackerInstall, PackerClean and co. Let me know if you have any feedback or questions!

File: lua/packer/display.lua
Comment (wbthomason) [inline lua/packer/display.lua:89]: Out of curiosity, why does this drop the check that `self.buf` is valid?

**Review by wbthomason — COMMENTED** [review]
This looks generally good, thanks! My only question before merging is why the check that the current display buffer remains valid was dropped? We're _probably_ ok without checking that, but I'd prefer to err on the side of caution (I don't think the check has significant performance impacts, and it is possible that some other plugin could interfere with the display buffer).

File: lua/packer/display.lua
Comment (fsouza) [reply lua/packer/display.lua:89]: It was an oversight, sorry about that. Let me fix it

Comment (fsouza) [reply lua/packer/display.lua]: Ok pushed a fix! :D

Comment (wbthomason) [reply lua/packer/display.lua]: Great, thank you!

**Review by wbthomason — APPROVED** [review]

**Pull request merged**
